bam_addrprg.c

```diff
@@ -374,7 +376,9 @@ static bool init(const parsed_opts_t* opts, state_t** state_out) {
     retval->input_header = sam_hdr_read(retval->input_file);
 
     retval->output_header = bam_hdr_dup(retval->input_header);
-    retval->output_file = sam_open_format(opts->output_name == NULL?"-":opts->output_name, "w", &opts->ga.out);
+    if (opts->output_name) // File format auto-detection
+        sam_open_mode(output_mode + 1, opts->output_name, NULL);
```

Comment:
We should check the return value of `sam_open_mode` and print an error message, if there are issues with detecting the format.

Reply:
This is only guessing the format based on the file suffix.  I don't think it should be an error if it fails.  Even a warning would be quite annoying, especially if the file being opened is "-".